### ISSUE TITLE
Add rule for double/dead/unused assignments

### ISSUE THREAD

**Issue by akx** [astral-sh/ruff#27980]
### Summary

Would be nice if Ruff detected double/dead/unused assignments such as 

```python
def convert(self, ..., **kwargs) -> ...:
    quantization = kwargs.get("quantization", False)
    quantization = True
    ...
```

([real-life source](https://github.com/NVIDIA-NeMo/Automodel/blob/d800383845f48a5e163f2325e5dac4a0b062ff7f/nemo_automodel/components/models/kimi_k25_vl/state_dict_adapter.py#L201-L202) that cost some $100s of rented GPU node time for me, but it is what it is).

## Proof of no detection

`foo.py`:

```
def convert(**kwargs) -> ...:
    quantization = kwargs.get("quantization", False)
    quantization = True
    return quantization + 1
```

```
$ uvx ruff@latest check --select=ALL --ignore=ANN,CPY,D --preview foo.py
All checks passed!
$
```

**Comment by MichaReiser** [comment]
This can get tricky once control flow is involved. 

ty already detects this, but editor only for now https://play.ty.dev/8501da55-632e-4cb6-bcf5-aec64fca37c1


We do plan to allow opt in to unused variables (with higher false positives) in the future.

**Comment by akx** [comment]
> This can get tricky once control flow is involved.

I have a (AI-assisted) 260-line rule for this that does this in a minimal way: 

> Rather than perform full control-flow analysis, the rule requires the two assignments to sit in one uninterrupted run of assignment statements within a single block, so nothing can branch, loop, or return in between.

Should I open a PR for scrutiny?
